genesyscloud/conversations_messaging_settings/resource_conversations_messaging_settings_schema.go

```diff
@@ -0,0 +1,161 @@
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Enabled", "Disabled"}, false),
+			},
+		},
+	}
+
+	contentSettingResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"story": {
+				Description: "Settings relating to facebook and instagram stories feature",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        storySettingResource,
+			},
+		},
+	}
+	storySettingResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"mention": {
+				Description: "Setting relating to Story Mentions",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        inboundOnlySettingResource,
+			},
+			"reply": {
+				Description: "Setting relating to Story Replies",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        inboundOnlySettingResource,
+			},
+		},
+	}
+	inboundOnlySettingResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"inbound": {
+				Description:  "Valid values: Enabled, Disabled.",
+				Optional:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"Enabled", "Disabled"}, false),
+			},
+		},
+	}
+)
+
+// SetRegistrar registers all of the resources, datasources and exporters in the package
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resourceName, ResourceConversationsMessagingSettings())
+	regInstance.RegisterDataSource(resourceName, DataSourceConversationsMessagingSettings())
+	regInstance.RegisterExporter(resourceName, ConversationsMessagingSettingsExporter())
+}
+
+func ResourceConversationsMessagingSettings() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud conversations messaging settings",
+
+		CreateContext: provider.CreateWithPooledClient(createConversationsMessagingSettings),
+		ReadContext:   provider.ReadWithPooledClient(readConversationsMessagingSettings),
+		UpdateContext: provider.UpdateWithPooledClient(updateConversationsMessagingSettings),
+		DeleteContext: provider.DeleteWithPooledClient(deleteConversationsMessagingSettings),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The messaging Setting profile name",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"content": {
+				Description: "Settings relating to message contents",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        contentSettingResource,
+			},
+			"event": {
+				Description: "Settings relating to events which may occur",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        eventSettingResource,
+			},
+		},
```

Comment:
If i try to create one of these with only the name field, I see this error

```json
{
  "message": "Config field is required",
  "code": "invalid.messaging.setting",
  "status": 400,
  "messageWithParams": "Config field is required",
  "contextId": "2252da7f-e244-4a8a-a6a6-5ef3186da8d5",
  "details": [],
  "errors": []
}
```

I don't see any field "config", but it looks like it works if you provide either `content` or `event` (and the nested values until you eventually provide a value for `inbound` or `outbound`). I would use the attribute `ExactlyOneOf`. There is an example of it being used in `resource_genesyscloud_outbound_contactlistfilter_schema.go`